base_layer/core/src/transaction.rs

```diff
@@ -84,10 +187,111 @@ pub struct TransactionKernel {
     /// Remainder of the sum of all transaction commitments. If the transaction
     /// is well formed, amounts components should sum to zero and the excess
     /// is hence a valid public key.
-    pub excess: Commitment,
+    pub excess: Option<Commitment>,
     /// The signature proving the excess is a valid public key, which signs
     /// the transaction fee.
-    pub excess_sig: Signature,
+    pub excess_sig: Option<Signature>,
 }
 
-pub type BlindingFactor = [u8; 32];
+/// Implementation of the transaction kernel
+impl TransactionKernel {
+    /// Creates an empty transaction kernel
+    pub fn empty() -> TransactionKernel {
+        TransactionKernel { features: KernelFeatures::empty(), fee: 0, lock_height: 0, excess: None, excess_sig: None }
+    }
+
+    /// Build a transaction kernel with the provided fee
+    pub fn with_fee(self, fee: u64) -> TransactionKernel {
```

Comment:
using mut self is much more optimal than consuming self and building a new one

Reply:
using mut self as suggested.